### PR TITLE
feat: alert rename interaction

### PR DESCRIPTION
**Pull request by ahmadshaheer**
### Summary
added rename interaction for alert, previously we would redirect to alert overview tab to edit the name. with this change, clicking on rename will open a modal and we can rename the alert. it also supports enter and escape key shortcuts to submit / cancel. 

#### Related Issues / PR's
close https://github.com/SigNoz/engineering-pod/issues/1899

#### Screenshots

https://github.com/user-attachments/assets/1d64a1e9-a616-4b63-b5f3-c5adca3d4ba5

NA

#### Affected Areas and Manually Tested Areas
Affected area:
- Alert Details Page ((Alert overview / Edit) Alert and history tabs)

Manually tested areas:
- Alert overview tab -> rename
- Alert history  tab -> rename

----

> [!IMPORTANT]
> Introduces modal-based alert renaming with keyboard shortcuts, updating components and hooks for improved user interaction.
> 
>   - **Behavior**:
>     - Adds modal-based rename functionality in `ActionButtons.tsx`, replacing redirect.
>     - Supports Enter and Escape keys for submit/cancel in `RenameModal.tsx`.
>   - **Components**:
>     - New `RenameModal` component for renaming alerts.
>     - Updates `AlertActionButtons` to open `RenameModal` and handle renaming logic.
>   - **Hooks**:
>     - Adds `useAlertRuleUpdate` in `hooks.tsx` for updating alert names.
>     - Modifies `useAlertRuleDelete`, `useAlertRuleDuplicate`, and `useAlertRuleStatusToggle` to integrate with new rename functionality.
>   - **Constants**:
>     - Adds `UPDATE_ALERT_RULE` to `REACT_QUERY_KEY` in `reactQueryKeys.ts`.
>     - Adds `alert` field to default alert configurations in `defaults.ts`.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=SigNoz%2Fsignoz&utm_source=github&utm_medium=referral)<sup> for c985d7f2cd7d4d09b180a0c5c34d3f779e459444. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

**Comment by vikrantgupta25** [comment]
build is failing..

**Comment by ahmadshaheer** [comment]
> build is failing..

Fixed, @vikrantgupta25 :slightly_smiling_face:

File: frontend/src/types/api/alerts/def.ts [hunk 1]
```diff
@@ -19,7 +19,7 @@ export const defaultSeasonality = 'hourly';
 export interface AlertDef {
 	id?: number;
 	alertType?: string;
-	alert?: string;
+	alert: string;
```
Comment (SagarRajput-7) [inline frontend/src/types/api/alerts/def.ts:22]: intentional?

**Review by SagarRajput-7 — APPROVED** [review]

**Review by SagarRajput-7 — APPROVED** [review]

File: frontend/src/types/api/alerts/def.ts (hunk 1 above)
Comment (ahmadshaheer) [reply frontend/src/types/api/alerts/def.ts:22]: Yes, I believe the alert name should be compulsory. I was getting a type error if we'd keep it optional.

File: frontend/src/pages/AlertDetails/AlertHeader/ActionButtons/RenameModal.tsx [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+}: Props): JSX.Element {
+	const inputRef = useRef<InputRef>(null);
+
+	useEffect(() => {
+		if (isOpen && inputRef.current) {
+			inputRef.current.focus();
+		}
+	}, [isOpen]);
+
+	const buttonProps = {
```
Comment (YounixM) [inline frontend/src/pages/AlertDetails/AlertHeader/ActionButtons/RenameModal.tsx:32]: Why this approach ?

Comment (ahmadshaheer) [reply frontend/src/pages/AlertDetails/AlertHeader/ActionButtons/RenameModal.tsx:32]: I wanted to make it DRYer.

**Review by SagarRajput-7 — APPROVED** [review]

**Review by YounixM — APPROVED** [review]

**Review by SagarRajput-7 — APPROVED** [review]

**Review by YounixM — APPROVED** [review]

**Comment by YounixM** [comment]
Updated the label to docs shipped as the required step reporting was not updating

**Pull request merged**